src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs

```diff
@@ -39,17 +29,22 @@ public class OpenIdConnectEvents : RemoteAuthenticationEvents, IOpenIdConnectEve
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        public Func<RedirectContext, Task> OnRedirectToAuthenticationEndpoint { get; set; } = context => Task.FromResult(0);
+        public Func<RedirectContext, Task> OnRedirectToIdentityProvider { get; set; } = context => Task.FromResult(0);
 
         /// <summary>
         /// Invoked before redirecting to the identity provider to sign out.
         /// </summary>
-        public Func<RedirectContext, Task> OnRedirectToEndSessionEndpoint { get; set; } = context => Task.FromResult(0);
+        public Func<RedirectContext, Task> OnRedirectToIdentityProviderForSignOut { get; set; } = context => Task.FromResult(0);
 
         /// <summary>
         /// Invoked after "authorization code" is redeemed for tokens at the token endpoint.
         /// </summary>
-        public Func<TokenResponseReceivedContext, Task> OnTokenResponseReceived { get; set; } = context => Task.FromResult(0);
+        public Func<TokenResponseMessageReceivedContext, Task> OnTokenResponseMessageReceived { get; set; } = context => Task.FromResult(0);
+
+        /// <summary>
+        /// Invoked when an IdToken has been validated and produced an AuthenticationTicket.
+        /// </summary>
+        public Func<TokenValidatedContext, Task> OnTokenValidated { get; set; } = context => Task.FromResult(0);
```

Comment:
> Are you referring to OnTokenValidated?

Yes.

> Also: what do you mean by validating the access token?

When receiving both an identity and an access token, Wilson validates them both in the same call (by computing the access token hash and comparing it to the `at_hash` claim extracted from the identity token). That's why I think `OnTokenValidated` seems too restrictive, as it doesn't simply validate the `id_token` but covers more things.